src/main/java/org/folio/service/caches/MappingParametersCache.java

```diff
@@ -4,45 +4,55 @@
 import com.github.benmanes.caffeine.cache.Caffeine;
 import io.vertx.core.Future;
 import io.vertx.core.Vertx;
-import io.vertx.core.http.HttpMethod;
-import io.vertx.core.json.Json;
-import org.apache.http.HttpStatus;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
-
 import org.folio.OrganizationCollection;
 import org.folio.processing.mapping.defaultmapper.processor.parameters.MappingParameters;
+import org.folio.rest.core.RestClient;
+import org.folio.rest.core.models.RequestContext;
+import org.folio.rest.core.models.RequestEntry;
 import org.folio.rest.util.OkapiConnectionParams;
-import org.folio.rest.util.RestUtil;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.TimeUnit;
 
+import static java.util.Objects.nonNull;
+import static java.util.stream.Collectors.toList;
+import static org.folio.service.orders.utils.HelperUtils.collectResultsOnSuccess;
+
 /**
  * The class responsible for caching {@link MappingParameters}
  */
 @Component
 public class MappingParametersCache {
   private static final Logger LOGGER = LogManager.getLogger();
-  public static final String ORGANIZATIONS = "/organizations/organizations?limit=%s";
 
-  @Value("${orders.cache.mapping.parameters.settings.limit:1000}")
-  public int settingsLimit;
+  private static final String ORGANIZATIONS = "/organizations/organizations";
+
+  private static final int MAX_CHUNK_SIZE = 20;
```

Comment:
I don't think we need this param at all. Lets make one request with set limit and check the total - if total is more than limit, make a required number of requests additionally